Design/spectrum_number_and_workspace_index_abstraction.md

```diff
@@ -0,0 +1,269 @@
+# Indexing Abstraction Layer
```

Comment:
We can rename the concept and keep the naming in the UI the same (if it gets not banished). As I wrote somewhere, what the UI sees is a global index, which in any case for an MPI run is not that same as the current "workspace index".